wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorImpl.java

```diff
@@ -244,6 +247,16 @@ public void run() {
       }
     } catch (final InterruptedException intEx) {
       // do nothing; exit thread
+      LOGGER.warning(
+          () -> Messages.get(
+              "MonitorImpl.interruptedExceptionDuringMonitoring",
+              new Object[] {this.hostSpec.getHost(), intEx.getMessage()}));
+    } catch (final Exception ex) {
+      // do nothing; exit thread
+      LOGGER.warning(
```

Comment:
I think we need to change code so such unhandled exceptions to be logged but the monitoring thread/loop keeps running.